server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -382,7 +382,8 @@ class Serializer @Inject() (mailboxIdFactory: MailboxId.Factory) {
   private implicit val textBodyWrites: Writes[TextBody] = Json.valueWrites[TextBody]
   private implicit val htmlBodyWrites: Writes[HtmlBody] = Json.valueWrites[HtmlBody]
 
-  implicit val vacationResponseWrites: Writes[VacationResponse] = Json.writes[VacationResponse]
+  implicit def vacationResponseWrites(properties: Set[String]): Writes[VacationResponse] = Json.writes[VacationResponse]
+    .transform((o: JsObject) => JsObject(o.fields.filter(entry => properties.contains(entry._1))))
```

Comment:
Rebase on https://github.com/linagora/james-project/pull/3698 to be improving the syntax ;-)